blueprints/ember-electron/index.js

```diff
@@ -27,122 +29,127 @@ module.exports = class EmberElectronBlueprint extends Blueprint {
     return entityName;
   }
 
-  afterInstall(/* options */) {
-    let logger = new Logger(this);
+  beforeInstall() {
+    if (fs.existsSync(electronProjectPath)) {
+      return Promise.reject(
+        new SilentError([
+          `Cannot create electron-forge project at './${electronProjectPath}'`,
+          `because a file or directory already exists there. Please remove/rename`,
+          `it and run the blueprint again: 'ember generate ember-electron'.`
+        ].join(' '))
+      );
+    }
 
-    return this._checkForgeConfig(logger)
-      .then(() => this._installElectronTooling(logger))
-      .then(() => this._createResourcesDirectories(logger))
-      .then(() => this._ensureForgeConfig(logger));
+    if (fs.existsSync('ember-electron')) {
+      this.ui.writeLine(chalk.yellow([
+        `\n'ember-electron' directory detected -- this looks like an ember-electron`,
+        `v2 project. Setting up an updated project will not be destructive, but you`,
+        `should read the upgrading documentation at ${upgradingUrl}.\n`
+      ].join(' ')));
+    }
   }
 
-  _checkForgeConfig(logger) {
-    const readJson = denodeify(fs.readJson);
-
-    let packageJsonPath = path.join(this.project.root, 'package.json');
-
-    return readJson(packageJsonPath)
-      .then((packageJson) => {
-        let hasForgeConfig = false;
-        let message;
-
-        try {
-          hasForgeConfig = packageJson.config.forge !== undefined;
-        } catch(err) {
-          // no-op
-        }
+  async afterInstall() {
+    await this.updateTravisYml();
+    await this.updateEslintIgnore();
+    await this.createElectronProject();
+  }
 
-        this.hasForgeConfig = hasForgeConfig;
-        message = `Project ${hasForgeConfig ? 'has' : 'needs'} forge config`;
+  async updateTravisYml() {
+    if (!fs.existsSync('.travis.yml')) {
+      this.ui.writeLine(chalk.yellow([
+        `\nNo .travis.yml found to update. For info on manually updating your CI'`,
+        `'config read ${ciUrl}'\n`
+      ].join(' ')));
+      return;
+    }
 
-        logger.message(message);
-      });
+    this.ui.writeLine(chalk.green('Updating .travis.yml'));
+
+    try {
+      let contents = await readFile('.travis.yml');
+      let yawn = new YAWN(contents.toString());
+
+      let doc = yawn.json;
```

Comment:
We should definitely explore using the built in ember-cli methods for modifying files https://ember-cli.com/api/classes/Blueprint.html#method_insertIntoFile

Reply:
Yeah, I switched to that on your suggestion for `.eslintignore`, but I'm more nervous about `.travis.yml`. Since the document has a structure that's so much more complex than an ignore file, I was thinking it would be better to use an AST parser rather than trying to handle all the various ways users might have modified their configs using text-only parsing.